platform-ajax/src/main/typescript/com/softicar/platform/ajax/engine/event/KeyboardEvents.ts

```diff
@@ -0,0 +1,144 @@
+
+class KeyboardEventManager {
+	private readonly handlers = new Map<string, KeyboardEventHandler>();
+
+	public setListenToKey(node: HTMLElement, eventName: string, enabled: boolean) {
+		let key = this.getKey(eventName);
+		this.getKeyHandler(node).setListenTo(key, eventName, enabled);
+	}
+	
+	public setFireOnKeyUp(node: HTMLElement, eventName: string, enabled: boolean) {
+		let key = this.getKey(eventName);
+		this.getKeyHandler(node).setFireOnKeyUp(key, enabled);
+	}
+	
+	public setPreventDefaultBehavior(node: HTMLElement, eventName: string, enabled: boolean) {
+		let key = this.getKey(eventName);
+		this.getKeyHandler(node).setPreventDefault(key, enabled);
+	}
+	
+	public setCssClassOnKeyDown(node: HTMLElement, eventName: string, cssTargetNode: HTMLElement, cssClassNames: string) {
+		let key = this.getKey(eventName);
+		this.getKeyHandler(node).setCssClassApplier(key, new CssClassApplier(cssTargetNode, cssClassNames));
+	}
+
+	private getKeyHandler(node: HTMLElement) {
+		let handler = this.handlers.get(node.id);
+		if(!handler) {
+			handler = new KeyboardEventHandler(node).install();
+			this.handlers.set(node.id, handler);
+		}
+		return handler;
+	}
+	
+	private getKey(eventName: string): number {
+		switch(eventName) {
+			case 'ENTER':  return DOM_VK_ENTER;
+			case 'ESCAPE': return DOM_VK_ESCAPE;
+			case 'SPACE':  return DOM_VK_SPACE;
+			case 'TAB':    return DOM_VK_TAB;
+		}
+		throw new Error("Internal error: Unsupported keyboard event name.");
+	}
+}
+
+const KEYBOARD_EVENT_MANAGER = new KeyboardEventManager();
+
+class KeyboardEventHandler {
```

Comment:
Apparently, there's a way to emulate private classes in TS:
https://stackoverflow.com/a/17163506
...not exactly necessary here - but interesting nevertheless.

Reply:
yes, indeed, but I think modularization is for another story